### PR TITLE
chore: Add error message for no provider found

### PR DESCRIPTION
**Pull request by wlee221**
#### Description of changes

Whenever customer are trying to use `useAuthenticator` without wrapping their App with `Authenticator.Provider`, they'll get the cryptic error message 

```
TypeError: Cannot read properties of undefined (reading 'send')
```

This PR adds a check for that and adds a more dev-friendly message.

#### Issue #, if available

#1651: dev confusion with errors when no provider is added/

#### Description of how you validated changes

No runtime behavior changes unless the app is already on faulty configuration. I might have missed edge cases like `Authenticator.Provider` not being ready by the time `useAuthenticator` first runs, but e2e test should catch that.

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [ ] Tests are updated
- [ ] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/react/src/components/Authenticator/hooks/useAuthenticator/index.tsx [hunk 1]
```diff
@@ -89,6 +89,16 @@ export type Selector = (context: AuthenticatorContext) => Array<any>;
...
 export const useAuthenticator = (selector?: Selector) => {
   const { service } = React.useContext(AuthenticatorContext);
+
+  if (!service) {
+    console.error(
+      'No `Authenticator.Provider` was found above where `useAuthenticator` hook is. ' +
+        'Please ensure you wrap your App with `Authenticator.Provider` like so: ' +
+        'https://ui.docs.amplify.aws/components/authenticator#useauthenticator-hook'
+    );
+    return null;
```
Comment (wlee221) [inline packages/react/src/components/Authenticator/hooks/useAuthenticator/index.tsx:99]: Wasn't sure whether we should `throw` or just return `null` here. 

I went with `null` to avoid breaking customer apps unexpectedly. But on the other hand, it also makes sense to `throw` because not having `Provider` will break their app anyway. Up to discussion!

**Review by 0618 — APPROVED** [review]

**Review by joebuono — APPROVED** [review]

File: packages/react/src/components/Authenticator/hooks/useAuthenticator/index.tsx (hunk 1 above)
Comment (ErikCH) [reply packages/react/src/components/Authenticator/hooks/useAuthenticator/index.tsx:99]: 👍
